File: src/main/java/com/danyazero/model/ast/NodeList.java
package com.danyazero.model.ast;

import com.danyazero.utils.GenerationContext;

import java.util.List;

public class NodeList implements Node {
    private final List<Node> statements;

    public NodeList(List<Node> statements) {
        this.statements = statements;
    }

    @Override
    public void produce(GenerationContext ctx) {
        statements.forEach(node -> node.produce(ctx));
    }

    @Override
    public void resolveTypes(GenerationContext ctx) {
        statements.forEach(node -> node.resolveTypes(ctx));
    }

    public List<Node> getStatements() {
        return statements;
    }
}

File: src/main/kotlin/node/NodeList.kt
package com.danyazero.node

import com.danyazero.model.Node
import com.danyazero.utils.GenerationContext

class NodeList(
    val nodes: List<Node>
) : Node{
    override fun produce(ctx: GenerationContext) {
        nodes.forEach { node -> node.produce(ctx) }
    }

    override fun toString(): String {
        return "NodeList(nodes=$nodes)"
    }
}